include/deal.II/distributed/shared_tria.h

```diff
@@ -106,6 +113,39 @@ namespace parallel
                                          const std::vector< CellData< dim > > &cells,
                                          const SubCellData &subcelldata);
 
+      /**
+       * Return auxilary vector subdomain_id_with_artificial .
```

Comment:
You can make the policy class a `friend`. You just need to forward declare the policy class and then declare the class a friend. See how it's already done in `include/dofs/dof_handler.h`.

But I think the approach above with only one public function with a descriptive name is also useful. I can see more uses for this function.